src/Net/Http.fs

```diff
@@ -1327,9 +1327,9 @@ module internal HttpHelpers =
                     else read
 
             override x.CanRead = true
-            override x.CanSeek = false
+            override x.CanSeek = match length with | None -> false | Some _ -> true
             override x.CanWrite = false
-            override x.Length with get () = length
+            override x.Length with get () = length |> Option.defaultWith (fun () -> NotSupportedException() |> raise)
```

Comment:
What do you think about a more descriptive message like in `Stream`? https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?view=net-6.0#exceptions

Reply:
If flipped on this a couple of times. But now rethinking, I reckon a simple `failwith "One or more of the encompassed streams are not seekable and the length cannot be determined"`. We can probably do away with the `NotSupportedException` type as the `CombinedStream` is only internal to `FSharp.data`